README.md

```diff
@@ -172,245 +197,193 @@ Ipld::DagJson::serialize(
+    Prefix         Metadata            Encoding
+  ┌────┴───┐┌───────────┴──────────┐┌─────┴────┐
+  │        ││ ┏━━━━┓┏━━━━━━┓┏━━━━┓ ││ ┏━━━━━━┓ │
+  │  0x34  ││ ┃0xEC┃┃0x1200┃┃0x1B┃ ││ ┃0x6A77┃ │
+  │        ││ ┗━━▲━┛┗━━▲━━━┛┗━━▲━┛ ││ ┗━━━▲━━┛ │
+  └────────┘└────┼─────┼───────┼───┘└─────┼────┘
+                 │     │       │          │
+               ECDSA   │   Keccak-256    JWT
+                       │
+                     P-256
 ```
 
-| Segment                 | Hexadecimal | Unsigned Varint | Comment                         | 
-|-------------------------|-------------|-----------------|---------------------------------|
-| `ed25519-varsig-header` | `%xED`      | `%xED01`        | Ed25519 key [multicodec] prefix |
+### Varsig Prefix
 
-## 5.3 ECDSA
+The varsig prefix MUST be the [multicodec] value `0x34`.
 
-ECDSA defines a general mechanism over many elliptic curves. 
+### Signature Algorithm Metadata
 
-``` abnf
-ecdsa-varsig = ecdsa-varsig-header ecdsa-hash-algorithm encoding-info sig-bytes
+The signature algorithm field MUST consist of one or more unsigned varint segments. The signature algorithm is often the [multicodec] of the associated public key, but MAY be unique for the signature type. The code MAY live outside the multicodec table. The first segment MUST act as a discriminant for how many expected segments follow in the signature algorithm field.
 
-ecdsa-varsig-header = unsigned-varint
-ecdsa-hash-algorithm = unsigned-varint
-encoding-info = 1*unsigned-varint
-sig-bytes = *OCTET
+``` abnf
+varsig-signature-algorithm
+  = %xB1   bls-public-key-curve        multihash-header ; BLS
+  / %xEC   ecdsa-curve recovery-parity multihash-header ; ECDSA
+  / %xED   eddsa-curve                 multihash-header ; EdDSA
+  / %x1205 rsa-size                    multihash-header ; RSASSA-PKCS #1 v1.5
 ```
 
-Here are a few examples encoded as varsig:
+### Payload Encoding Metadata
 
-### 5.3.1 Example: ES256
+The [IPLD] data model is encoding agnostic by design. This is very convenient in many applications, such as making for very convenient conversions between types for transmission versus encoding. Unfortunately signatures require signing over specific bytes, and thus over a specific encoding of the data. To facilitate this, the type `varsig-encoding-info` MUST be used:
 
 ``` abnf
-es256-varsig = es256-varsig-header es256-hash-algorithm encoding-info sig-bytes
-
-es256-varsig-header = %x1200 ; P-256 multicodec prefix
-es256-hash-algorithm = %x12 ; SHA2-256
-encoding-info = 1*unsigned-varint
-sig-bytes = 64(OCTET)
+varsig-encoding-info
+  = %x5F                        ; Single verbatim payload (without key)
+  / %x70                        ; DAG-PB multicodec prefix
+  / %x71                        ; DAG-CBOR multicodec prefix
+  / %x0129                      ; DAG-JSON multicodec prefix
+  / %x6A77                      ; Canonicalized JWT
+  / %xE191 varsig-encoding-info ; EIP-191 "personal sign"
```

Comment:
```suggestion
  / 0xd191 ; EIP-191 "personal sign"
```

Reply:
I think we still need to know how to format the payload, right? IIRC the EIP-191 just sticks the domain separator to the fully serialized payload, or is there some detail I'm forgetting?

Reply:
ah i see what you mean, you are right! we may even need to put encoding and serialization.
need to check if wallets support sending bytes or only strings